components/motor/gpiostepper/gpiostepper.go

```diff
@@ -283,35 +270,53 @@ func (m *gpioStepper) GoFor(ctx context.Context, rpm, revolutions float64, extra
 	return m.opMgr.WaitTillNotPowered(ctx, time.Millisecond, m, m.Stop)
 }
 
-func (m *gpioStepper) goForInternal(ctx context.Context, rpm, revolutions float64) error {
+func (m *gpioStepper) goForMath(ctx context.Context, rpm, revolutions float64) error {
 	if revolutions == 0 {
+		// go a large number of revolutions if 0 is passed in, at the desired speed
 		revolutions = 1000000.0
 	}
-	var d int64 = 1
-
-	if math.Signbit(revolutions) != math.Signbit(rpm) {
-		d = -1
-	}
-
-	revolutions = math.Abs(revolutions)
-	rpm = math.Abs(rpm) * float64(d)
 
 	if math.Abs(rpm) < 0.1 {
+		m.logger.Debug("motor (%s) speed less than .1 rev_per_min, stopping", m.motorName)
 		return m.Stop(ctx, nil)
 	}
 
 	m.lock.Lock()
 	defer m.lock.Unlock()
 
+	// set the direction pin here,not in the thread, so we're handling only one pin in the goroutine
+	var d int64 = 1
+	if math.Signbit(revolutions) != math.Signbit(rpm) {
+		d = -1
+		if err := m.dirPin.Set(ctx, false, nil); err != nil {
```

Comment:
I prefer the "old" way of doing this where all pins are set by the same thread. with this implementation, we end up with one thread manipulating steps and another thread manipulating direction and enable. it puts manipulation of these pins out of sync with each other.

Reply:
If you feel strongly about it, I'll change it back.